spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -234,6 +234,61 @@ public Table loadTable(Identifier ident, long timestamp) throws NoSuchTableExcep
     }
   }
 
+  @Override
+  public boolean tableExists(Identifier ident) {
+    try {
+      if (isPathIdentifier(ident)) {
+        loadFromPathIdentifier((PathIdentifier) ident);
+        return true;
+      } else {
+        boolean isExists = icebergCatalog.tableExists(buildIdentifier(ident));
+        if (isExists) {
+          return true;
+        }
+
+        // if the original load didn't work, try using the namespace as an identifier because
```

Comment:
Do we need this logic? I keep consistent with origin logic. Origin logic will reuse the method `loadTable`.

Reply:
Thanks. I have modified.